lib/web_ui/lib/src/engine/compositor/painting.dart

```diff
@@ -0,0 +1,199 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of engine;
+
+/// The implementation of [ui.Paint] used by the CanvasKit backend.
+class SkPaint implements ui.Paint {
```

Comment:
fwiw, having multiple implementations of Paint is going to hurt performance as it means everywhere that calls into a Paint now has to do a dynamic lookup.

Reply:
Not necessarily. Since `experimentalUseSkia` is `const`, dart2js should be able to figure out that only one non-abstract subclass of `Paint` is live, and avoid dynamic dispatch. I will double check this

Reply:
Just built the gallery with this change and it looks like dart2js has inlined the `SkPaint` constructor into every `new Paint()` call, so dynamic dispatch is avoided. Here's a sample function that creates a `Paint`:

![skpaint-verify](https://user-images.githubusercontent.com/1961493/70761310-9fb78a00-1d01-11ea-9dac-40d160cdc62c.png)